### PR TITLE
✨  add deepfence threatmapper

### PR DESCRIPTION
**Pull request by manuel-sommer**
see issue #9687

### CODE REVIEW

**Comment by shodanwashere** [comment]
I'm checking your code here, so you wrote the parser so that it can automatically identify if it's one of the 4 report types? All findings will therefore be aggregated together?

**Review by shodanwashere — APPROVED** [review]
4 different reports with different formats were given, and they can all be aggregated under one single report type.
LGTM!

**Comment by mtesauro** [comment]
@manuel-sommer @shodanwashere Just for awareness, there's some very new parser guidelines I wrote up at 
https://github.com/DefectDojo/django-DefectDojo/discussions/9690

In this specific case, I think this is fine because they are from the same tool (and type of tool) and have generally the same info expressed in different file format.

Would love to have you confirm that since shodanwashere appears to be a user of this tool.

**Comment by manuel-sommer** [comment]
Sounds legit, thank you for the hint @mtesauro. I would be ok with both ways here: approve how it is right now or I can go with your policy and make 4 parsers out of it, depends on your review.

File: dojo/tools/deepfence_threatmapper/parser.py
Comment (cneill) [inline dojo/tools/deepfence_threatmapper/parser.py:112]: Since the de-duplication hashcode includes the description, including the scan ID likely doesn't make sense if it is unique every time the tool runs.

File: dojo/tools/deepfence_threatmapper/parser.py
Comment (cneill) [inline dojo/tools/deepfence_threatmapper/parser.py:110]: The spacing here seems to mess up the Markdown parser when there's no value included after the `**`. I think all of these properties should be reworked like so:
```suggestion
                description += "**cloud_account_id:** " + cloud_account_id + "\n"
```

Screenshot 2024-03-08 at 12 53 13

Screenshot 2024-03-08 at 12 56 04

**Review by cneill — CHANGES_REQUESTED** [review]
A couple thoughts on this while we wait for a reply to [this question](https://github.com/DefectDojo/django-DefectDojo/issues/9687#issuecomment-1984922055) about how the tool actually formats its output

**Review by cneill — DISMISSED** [review]

**Comment by manuel-sommer** [comment]
Closed because of the failed unittest, reopening now.

File: dojo/tools/deepfence_threatmapper/parser.py
Comment (manuel-sommer) [inline dojo/tools/deepfence_threatmapper/parser.py:74]: @cneill: I have no clue here why I need this check, but otherwise, two additional findings will be added with all values "None".

**Comment by manuel-sommer** [comment]
Reminder @cneill

**Comment by manuel-sommer** [comment]
Reminder to review @cneill and @mtesauro

File: dojo/tools/deepfence_threatmapper/parser.py
Comment (Maffooch) [inline dojo/tools/deepfence_threatmapper/parser.py:1]: Can these elif branches be broken into functions?

Comment (manuel-sommer) [reply dojo/tools/deepfence_threatmapper/parser.py:1]: done, good point @Maffooch

**Comment by manuel-sommer** [comment]
Any updates on this?

**Comment by manuel-sommer** [comment]
@Maffooch and @cneill can you review this please?

**Comment by manuel-sommer** [comment]
Hi @mtesauro, 

it would be nice if you could review this. I just rebased this PR to also include the settings.disp sha hash sum.

**Review by mtesauro — APPROVED** [review]
Approved

**Comment by shodanwashere** [comment]
> @manuel-sommer @shodanwashere Just for awareness, there's some very new parser guidelines I wrote up at #9690
> 
> In this specific case, I think this is fine because they are from the same tool (and type of tool) and have generally the same info expressed in different file format.
> 
> Would love to have you confirm that since shodanwashere appears to be a user of this tool.

I've read the parser guidelines and I can see where you're coming from. Thankfully, the guidelines that were written in that quarterly update make it so this multi-parser is an exception to the rule, so I don't have any comments to make on that note.

However, I'm also interested on how it would affect parsers for some other tools, like Harbor, which I use myself as well, and can encompass different scanning tools from different vendors, like AquaSec's Trivy and Quay's Clair scanners. Other than that, it seems good to me.

**Comment by cneill** [comment]
Looks like this one needs a rebase to update `.settings.dist.py.sha256sum`. The code looks good to me, so if the tests pass after a rebase I'll :+1:

**Comment by manuel-sommer** [comment]
Rebase done @cneill

File: dojo/settings/settings.dist.py
Comment (Maffooch) [inline dojo/settings/settings.dist.py:1487]: This looks like an oops

**Review by Maffooch — CHANGES_REQUESTED** [review]

File: dojo/settings/settings.dist.py
Comment (manuel-sommer) [reply dojo/settings/settings.dist.py:1487]: oh yeah, sorry for that

**Comment by manuel-sommer** [comment]
Done @Maffooch, you can continue the review.

**Comment by Maffooch** [comment]
Yep! Will approve once all tests are clear :)

**Review by grendel513 — APPROVED** [review]

**Review by cneill — APPROVED** [review]

**Comment by cneill** [comment]
Would you mind recalculating the hash and resolving the merge conflict @manuel-sommer ?

**Comment by mtesauro** [comment]
@manuel-sommer Mind one more rev on the sha256 sum for settings.dist.py? Order of PR merges put this one into merge conflict.

Thanks :pray:

**Comment by manuel-sommer** [comment]
Done @mtesauro and @cneill

**Pull request merged**
